libdevcore/db.h

```diff
@@ -32,62 +32,45 @@ namespace dev
 namespace db
 {
 
-// Transaction implements database transaction for a specific concrete database
-// implementation. The transaction implementation must call rollback in its
-// destructor if commit has not yet been called (making sure to catch any
-// exceptions).
-class Transaction
+// WriteBatchFace implements database write batch for a specific concrete
+// database implementation.
+class WriteBatchFace
 {
 public:
-	virtual ~Transaction() = default;
+	virtual ~WriteBatchFace() = default;
 
-	virtual void insert(Slice const& _key, Slice const& _value) = 0;
-	virtual void kill(Slice const& _key) = 0;
-
-	virtual void commit() = 0;
-	virtual void rollback() = 0;
+	virtual void insert(Slice _key, Slice _value) = 0;
+	virtual void kill(Slice _key) = 0;
 
 protected:
-	Transaction() = default;
+	WriteBatchFace() = default;
 	// Noncopyable
-	Transaction(const Transaction&) = delete;
-	Transaction& operator=(const Transaction&) = delete;
+	WriteBatchFace(const WriteBatchFace&) = delete;
+	WriteBatchFace& operator=(const WriteBatchFace&) = delete;
 	// Movable
-	Transaction(Transaction&&) = default;
-	Transaction& operator=(Transaction&&) = default;
+	WriteBatchFace(WriteBatchFace&&) = default;
```

Comment:
See the answer by Herb Sutter in the link you gave https://stackoverflow.com/a/14303116/1636308
What we are dealing with here is type 2, `WriteBatchFace` is going to be used only through pointers and references and it doesn't need to be movable. It's an interface anyway, making an interface movable doesn't  really make sense to me.

It doesn't even make `LevelDBWriteBatch` movable, it's still nonmovable now I think (has const memeber)
But doesn't need to anyway.

Reply:
That answer actually says nothing about making a type non-movable, only non-copyable. I changed it now regardless, but the question still stands. Also, I agree `const` members make move-assignment impossible, but not move-construction.

Reply:
_These do not need move construction or assignment for two reasons: They're not copyable, and they already have an even more efficient natural "move" operation -- you just copy/move the pointer to the object and the object itself doesn't have to move to a new memory location at all._